.gitignore

```diff
@@ -1,2 +1 @@
-Cargo.lock
```

Comment:
The general strategy is to commit `Cargo.lock` for binaries but not for libraries. Currently we don't commit any of them because they cause lots of merge conflicts during the rapid development phase of the project. So please don't remove this.

Reply:
Yeah i only removed this to test my theory of why travis was breaking, wasnt planning on leaving it in